internal/net/http/middleware/timeout.go

```diff
@@ -51,7 +51,7 @@ func (t *timeout) Wrap(h rest.Func) rest.Func {
 		// run the custom handler logic in go routine,
 		// report error to error channel
 		ech := make(chan error)
-		sch := make(chan int)
+		sch := make(chan int, 1)
```

Comment:
:memo:

Reply:
LGTM.
I think ech have same problem for future work we can change ech to 1 buffered channel